src/jit/lsra.cpp

```diff
@@ -6663,23 +6745,45 @@ bool LinearScan::isSecondHalfReg(RegRecord* regRec, Interval* interval)
 //    a lclVar was spilled after the block had been completed.
 void LinearScan::processBlockStartLocations(BasicBlock* currentBlock, bool allocationPass)
 {
-    unsigned    predBBNum         = blockInfo[currentBlock->bbNum].predBBNum;
-    VarToRegMap predVarToRegMap   = getOutVarToRegMap(predBBNum);
-    VarToRegMap inVarToRegMap     = getInVarToRegMap(currentBlock->bbNum);
+    // If we have no register candidates we should only call this method during allocation.
+
+    assert(enregisterLocalVars() || allocationPass);
+
+    if (!enregisterLocalVars())
+    {
+        // Just clear any constant registers and return.
+        for (regNumber reg = REG_FIRST; reg < ACTUAL_REG_COUNT; reg = REG_NEXT(reg))
+        {
+            RegRecord* physRegRecord = getRegisterRecord(reg);
+            Interval*  assignedInterval = physRegRecord->assignedInterval;
+
+            if (assignedInterval != nullptr)
+            {
+                assert(assignedInterval->isConstant);
+                physRegRecord->assignedInterval = nullptr;
+            }
+        }
+        INDEBUG(dumpLsraAllocationEvent(LSRA_EVENT_START_BB, nullptr, REG_NA, currentBlock));
+        return;
+    }
+
+    unsigned    predBBNum = blockInfo[currentBlock->bbNum].predBBNum;
+    VarToRegMap predVarToRegMap = getOutVarToRegMap(predBBNum);
+    VarToRegMap inVarToRegMap = getInVarToRegMap(currentBlock->bbNum);
     bool        hasCriticalInEdge = blockInfo[currentBlock->bbNum].hasCriticalInEdge;
 
-    const VARSET_TP* liveIn = &currentBlock->bbLiveIn;
+    VarSetOps::AssignNoCopy(compiler, currentLiveVars, VarSetOps::Intersection(compiler, registerCandidateVars, currentBlock->bbLiveIn));
 #ifdef DEBUG
     if (getLsraExtendLifeTimes())
     {
-        liveIn = &compiler->lvaTrackedVars;
+        VarSetOps::AssignNoCopy(compiler, currentLiveVars, compiler->lvaTrackedVars);
```

Comment:
Actually, it should just use `registerCandidateVars` (which is a strict subset of `lvaTrackedVars`)